src/deploy.ts

```diff
@@ -104,12 +110,9 @@ export async function deploy(
   let authError: null | "unauthenticated" | "forbidden" = null;
   try {
     if (apiKey) {
-      const invalidTiers = new Set(["basic", "enterprise", "public", "pro", "pro_enterprise"]);
       currentUser = await apiClient.getCurrentUser();
       // List of valid workspaces that can be used to create projects.
-      currentUser.workspaces = currentUser.workspaces.filter(
-        (w) => (w.role === "owner" || w.role === "member") && !invalidTiers.has(w.tier)
-      );
+      currentUser.workspaces = validWorkspaces(currentUser.workspaces);
```

Comment:
can we write this without mutation?

maybe:

```
({workspaces, ...currentUser} = await apiClient.getCurrentUser());
currentUser.workspaces = validWorkspaces(workspaces);
```

or

```suggestion
      currentUser = {...currentUser, workspaces: validWorkspaces(currentUser.workspaces)};
```

or make `validWorkspaces` consume and return a (different) user.